### PR TITLE
Fix #687: Accessibility on range slider with keyboard control

### LINKED ISSUE #687: Accessibility
Issues
=============

When we move a "two cursors slider" only with keyboard, interval is not correctly updated if one cursor passes over the other cursor (interval values should be inverted as they are when we cross cursors with a mouse).

### PR DESCRIPTION
**Pull request by jerrylow**
Fix issue #687 with keyboard control of range slider.

JSFiddle: https://jsfiddle.net/pgzp515m/

### CODE REVIEW

**Comment by jerrylow** [comment]
I am having issues running tests/phantomJS locally - not sure if anybody has any information on this. It's exactly this issue: https://github.com/gruntjs/grunt-lib-phantomjs/issues/89

File: src/js/bootstrap-slider.js [hunk 1]
```diff
@@ -1565,9 +1565,12 @@ const windowIsDefined = (typeof window === "object");
 				}
 
 				var val = this._state.value[handleIdx] + dir * this.options.step;
+				var percentage = (val / this.options.max) * 100;
+				this._state.keyCtrl = handleIdx;
 				if (this.options.range) {
-					val = [ (!handleIdx) ? val : this._state.value[0],
-						    ( handleIdx) ? val : this._state.value[1]];
+					this._adjustPercentageForRangeSliders(percentage);
+					val = [ (!this._state.keyCtrl) ? val : this._state.value[0],
```
Comment (rovolution) [inline src/js/bootstrap-slider.js]: could you break these array values into separate variables and then reference them within the array?

```js
const val1 = (!this._state.keyCtrl) ? val : this._state.value[0];
const val2 = this._state.keyCtrl) ? val : this._state.value[1];

val = [ val1, val2];
```

File: src/js/bootstrap-slider.js [hunk 2]
```diff
@@ -1565,9 +1565,12 @@ const windowIsDefined = (typeof window === "object");
 				}
 
 				var val = this._state.value[handleIdx] + dir * this.options.step;
+				var percentage = (val / this.options.max) * 100;
```
Comment (rovolution) [inline src/js/bootstrap-slider.js]: could you re-declare this as a `const`?

**Comment by jerrylow** [comment]
@rovolution - updated. Yes let me know if you find out more. I've reinstall phantomJS in different ways and different places - but to be honest don't think that's the cause of the problem. This was not a problem before I upgraded to Sierra.

File: src/js/bootstrap-slider.js [hunk 3]
```diff
@@ -1565,9 +1565,13 @@ const windowIsDefined = (typeof window === "object");
 				}
 
 				var val = this._state.value[handleIdx] + dir * this.options.step;
+				const percentage = (val / this.options.max) * 100;
+				this._state.keyCtrl = handleIdx;
 				if (this.options.range) {
-					val = [ (!handleIdx) ? val : this._state.value[0],
-						    ( handleIdx) ? val : this._state.value[1]];
+					this._adjustPercentageForRangeSliders(percentage);
+          const val1 = (!this._state.keyCtrl) ? val : this._state.value[0];
```
Comment (rovolution) [inline src/js/bootstrap-slider.js]: could you fix the spacing? thanks!

**Comment by jerrylow** [comment]
@rovolution done

**Comment by rovolution** [comment]
@jerrylow merged and published to 9.7.2

**Comment by rovolution** [comment]
thanks again for another great contribution!

**Pull request merged**